packages/@aws-cdk/app-staging-synthesizer-alpha/lib/default-staging-stack.ts

```diff
@@ -358,7 +367,15 @@ export class DefaultStagingStack extends Stack implements IStagingResources {
     }
 
     this.ensureFileRole();
-    const key = this.createBucketKey();
+
+    let key = undefined;
+    if (this.stagingBucketEncryption === s3.BucketEncryption.KMS || this.stagingBucketEncryption === undefined) {
+      if (this.stagingBucketEncryption === undefined) {
+        // default is KMS for backwards compatibility
+        this.stagingBucketEncryption = s3.BucketEncryption.KMS;
```

Comment:
Thanks @msambol. Give me a sec to figure this out, we don't necessarily have to honor backwards compatibility but I'm not recalling why we went with KMS as the default in the first place. Let me ask around

Reply:
Ok -- we are using KMS because that is officially AWS's best practice. So we're going to keep that as default as well.